src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter((oneAnimal) => oneAnimal !== animal);
```

Comment:
[CODE PERFORMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method without checking the presence of the animal in the array before filtering. The current implementation can be improved by removing the if condition on lines 31-32 and always running the filter after decreasing the health. This ensures that any animal with health less than or equal to 0 is removed from the array in a single operation, which is more efficient.